src/impl_methods.rs

```diff
@@ -124,7 +124,35 @@ impl<A, S, D> ArrayBase<S, D> where S: Data<Elem=A>, D: Dimension
         }
     }
 
-    /// Return an uniquely owned copy of the array
+    /// Return an uniquely owned copy of the array.
+    ///
+    /// For performance, the memory layout of the resulting array is arbitrary.
+    /// If you need a particular layout, you can allocate a new array with the
+    /// desired memory layout and [`.assign()`](#method.assign) the data.
+    /// Alternatively, you can collect an iterator, like this for a result in
+    /// standard layout:
```

Comment:
This can be fine as a contract, but I don't really agree with it. The layout of the output should be the same as that of the input, if possible. Then we restrict and say it's only possible if the input is contiguous (which is more or less true). Then yes, that's not entirely correct with the current implementation, since not all contiguous arrays are covered right now (but most..).

Reply:
Okay, how about

> If the input array is contiguous and its strides are positive, then the output array will have the same memory layout. Otherwise, the layout of the output array is unspecified. If you need a particular layout…

As far as I can tell, this description matches the current implementation. Once we add support for negative strides in `.is_contiguous()`, we can remove the "strides are positive" constraint.

Reply:
Yes that's good